File: src/main/java/com/wgzhao/addax/admin/repository/EtlJourRepo.java
package com.wgzhao.addax.admin.repository;

import com.wgzhao.addax.admin.model.EtlJour;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface EtlJourRepo
        extends JpaRepository<EtlJour, Long> {

    void deleteAllByTid(long tableId);

    @Query(value = """
            select error_msg from etl_jour
            where tid = :tableId and status = false
            order by id desc
            limit 1
            """, nativeQuery = true)
    String findLastError(long tableId);
    // 可根据需要添加自定义查询方法
}

File: src/main/kotlin/com/wgzhao/addax/admin/repository/EtlJourRepo.kt
package com.wgzhao.addax.admin.repository

import com.wgzhao.addax.admin.model.EtlJour
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface EtlJourRepo

    : JpaRepository<EtlJour?, Long?> {
    fun deleteAllByTid(tableId: Long)


    fun findFirstByTidAndStatusIsFalse(tableId: Long): EtlJour?

    fun findFirstByTidAndStatusIsFalseOrderByIdDesc(tableId: Long): EtlJour?
}
